### PR TITLE
Fix #3803: Bazel/Gradle Support on Kitkat [BLOCKED on #3804]

### LINKED ISSUE #3803: Bazel/Gradle Support on Kitkat
**Describe the bug**
Currently, the Bazel build apk does not work on kitkat (API 19)

**To Reproduce**
1. Build apk using Bazel and install on API level 19.
2. App crash

**Expected behavior**
The app should run correctly without any crashes.

**Additional Context**
As per AlphaMR3, if it takes longer to bring Bazel support, use Gradle as a build system to run APK on KitKat version.

### PR DESCRIPTION
**Pull request by anandwana001**
## Explanation

Fix #3803

APK supports on Kitkat when built using Gradle.

APK is building using Bazel, but crash when run on Kitkat version with the following errors:
1. Not able to bind the vector tag on the splash screen
2. Unable to start service androidx.work.impl.background.systemalarm.SystemAlarmService@9d1ab7c0

The permission issue can be solved using adding permission to manifest 
<uses-permission android:name="android.permission.WAKE_LOCK" />, but need to figure out the exact reason why KitKat needs this permission. 

## Checklist

- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The PR explanation includes the words "Fixes #bugnum: ..." (or "Fixes part of #bugnum" if the PR only partially fixes an issue).
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary auto-generated code from Android Studio.
- [x] The PR is made from a branch that's **not** called "develop".
- [x] The PR is made from a branch that is up-to-date with "develop".
- [x] The PR's branch is based on "develop" and not on any other branch.
- [x] The PR is **assigned** to an appropriate reviewer in both the **Assignees** and the **Reviewers** sections.

### CODE REVIEW

File: app/BUILD.bazel [hunk 1]
```diff
@@ -449,6 +449,13 @@ BINDING_ADAPTERS = [
 # keep sorted
 SNAP_HELPER = ["src/main/java/org/oppia/android/app/recyclerview/StartSnapHelper.kt"]
 
+EXCLUDE_OPPIA_APPLICTATION = [
+    "src/main/java/org/oppia/android/app/application/OppiaApplication.kt",
+    "src/main/java/org/oppia/android/app/application/ApplicationInjector.kt",
+    "src/main/java/org/oppia/android/app/application/ApplicationInjectorProvider.kt",
+    "src/main/java/org/oppia/android/app/application/ApplicationComponent.kt"
+]
```
Comment (anandwana001) [inline app/BUILD.bazel]: Not including these files as it contains data module items that are required for full app support.

Comment (anandwana001) [reply app/BUILD.bazel]: The changes in this file don't look correct in terms of formatting, once we get the final solution will clean format these changes as well.

File: app/src/main/java/org/oppia/android/app/application/ApplicationInjectorKitkat.kt [hunk 2]
```diff
@@ -0,0 +1,6 @@
+package org.oppia.android.app.application
+
+import org.oppia.android.util.data.DataProvidersInjector
+
+/** Injector for application-level dependencies that can't be directly injected where needed. */
+interface ApplicationInjectorKitkat
```
Comment (anandwana001) [inline app/src/main/java/org/oppia/android/app/application/ApplicationInjectorKitkat.kt]: Will add EOF

File: app/src/main/java/org/oppia/android/app/application/ApplicationInjectorProviderKitkat.kt [hunk 3]
```diff
@@ -0,0 +1,6 @@
+package org.oppia.android.app.application
+
+/** Provider for [ApplicationInjector]. The application context will implement this interface. */
+interface ApplicationInjectorProviderKitkat {
+  fun getApplicationInjectorKitkat(): ApplicationInjectorKitkat
+}
```
Comment (anandwana001) [inline app/src/main/java/org/oppia/android/app/application/ApplicationInjectorProviderKitkat.kt]: Will add EOF

File: app/src/main/java/org/oppia/android/app/application/ApplicationKitkatComponent.kt [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+    FirebaseLogUploaderModule::class, PracticeTabModule::class,
+    PlatformParameterModule::class, ExplorationStorageModule::class,
+    DeveloperOptionsStarterModule::class, DeveloperOptionsModule::class,
+    PlatformParameterSyncUpWorkerModule::class, NetworkConnectionUtilDebugModule::class,
+    // TODO(#59): Remove this module once we completely migrate to Bazel from Gradle as we can then
+    //  directly exclude debug files from the build and thus won't be requiring this module.
+    NetworkConnectionDebugUtilModule::class
+  ]
+)
+interface ApplicationKitkatComponent : ApplicationInjectorKitkat {
```
Comment (anandwana001) [inline app/src/main/java/org/oppia/android/app/application/ApplicationKitkatComponent.kt]: New application components which doesn't have any data module items.

File: domain/src/main/java/org/oppia/android/domain/platformparameter/syncup/PlatformParameterSyncUpWorker.kt [hunk 5]
```diff
@@ -117,12 +111,11 @@ class PlatformParameterSyncUpWorker private constructor(
       exceptionsController.logNonFatalException(e)
       Result.failure()
     }
-  }
+  }*/
 
   /** Creates an instance of [PlatformParameterSyncUpWorker] by properly injecting dependencies. */
   class Factory @Inject constructor(
     private val platformParameterController: PlatformParameterController,
-    private val platformParameterService: PlatformParameterService,
```
Comment (anandwana001) [inline domain/src/main/java/org/oppia/android/domain/platformparameter/syncup/PlatformParameterSyncUpWorker.kt:125]: domain module file injecting data module item. 
Removing it for temporary.

Comment (anandwana001) [reply domain/src/main/java/org/oppia/android/domain/platformparameter/syncup/PlatformParameterSyncUpWorker.kt:125]: @ARJUPTA 
Could we separate it out in such a way that the domain modules do not use it directly?

**Comment by BenHenning** [comment]
@anandwana001 this isn't introducing bundle support yet since it isn't using android_application--can you try out that target first to see if it works before separating out data module?

**Comment by BenHenning** [comment]
@anandwana001 now that #3750 exists please update this to be branched off of that & update the approach to change the flavors defined in build_flavors.bzl, instead.

**Comment by anandwana001** [comment]
The current implementation in this PR works when built and install using gradle. 

Getting following error with Bazel. 
```
E/AndroidRuntime: FATAL EXCEPTION: main
    Process: org.oppia.android, PID: 4098
    java.lang.RuntimeException: Unable to instantiate application org.oppia.android.app.application.OppiaApplication: java.lang.ClassNotFoundException: Didn't find class "org.oppia.android.app.application.OppiaApplication" on path: DexPathList[[zip file "/data/app/org.oppia.android-1.apk"],nativeLibraryDirectories=[/data/app-lib/org.oppia.android-1, /vendor/lib, /system/lib]]
        at android.app.LoadedApk.makeApplication(LoadedApk.java:516)
        at android.app.ActivityThread.handleBindApplication(ActivityThread.java:4317)
        at android.app.ActivityThread.access$1500(ActivityThread.java:135)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1256)
        at android.os.Handler.dispatchMessage(Handler.java:102)
        at android.os.Looper.loop(Looper.java:136)
        at android.app.ActivityThread.main(ActivityThread.java:5017)
        at java.lang.reflect.Method.invokeNative(Native Method)
        at java.lang.reflect.Method.invoke(Method.java:515)
        at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:779)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:595)
        at dalvik.system.NativeStart.main(Native Method)
     Caused by: java.lang.ClassNotFoundException: Didn't find class "org.oppia.android.app.application.OppiaApplication" on path: DexPathList[[zip file "/data/app/org.oppia.android-1.apk"],nativeLibraryDirectories=[/data/app-lib/org.oppia.android-1, /vendor/lib, /system/lib]]
        at dalvik.system.BaseDexClassLoader.findClass(BaseDexClassLoader.java:56)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:497)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:457)
        at android.app.Instrumentation.newApplication(Instrumentation.java:975)
        at android.app.LoadedApk.makeApplication(LoadedApk.java:511)
        at android.app.ActivityThread.handleBindApplication(ActivityThread.java:4317) 
        at android.app.ActivityThread.access$1500(ActivityThread.java:135) 
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1256) 
        at android.os.Handler.dispatchMessage(Handler.java:102) 
        at android.os.Looper.loop(Looper.java:136) 
        at android.app.ActivityThread.main(ActivityThread.java:5017) 
        at java.lang.reflect.Method.invokeNative(Native Method) 
        at java.lang.reflect.Method.invoke(Method.java:515) 
        at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:779) 
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:595) 
        at dalvik.system.NativeStart.main(Native Method) 
```

**Comment by BenHenning** [comment]
@anandwana001 do we have an issue tracking the latest KitKat issue (the SVG issue)?

Also, can you please update the PR title & description to more correctly describe what this PR is actually doing? We now have AAB support checked in, so the current title is no longer correct.

**Comment by anandwana001** [comment]
closing this as this is handled in other pr.

**Pull request closed without merging**